src/Avalonia.Visuals/Rendering/SceneGraph/RectangleNode.cs

```diff
@@ -30,7 +30,7 @@ internal class RectangleNode : BrushDrawOperation
             double radiusX,
             double radiusY,
             IDictionary<IVisual, Scene> childScenes = null)
-            : base(rect, transform, pen)
+            : base(rect.Inflate((pen?.Thickness ?? 0) / 2), transform, pen)
```

Comment:
If two adjacent rects are drawn do the dirty rects align? If not is that the wanted behavior?

Reply:
I dont know, @grokys probably has a better idea about what needs to be done here. this change is just maintaining the existing behavior.

Reply:
@Gillibald the dirtyrects are snapped to pixels, and they also union if they partially overlap.

Reply:
@grokys has resolved this.